src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1080,6 +1077,11 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger? errorLogger, Cancella
                     }
                 }
 
+                if (Arguments.ParseOptions.Features.ContainsKey("debug-determinism"))
```

Comment:
This is existing code, but seems to be lacking test coverage (can't find "debug-determinism" anywhere else)

Reply:
This feature flag is used in the determinism PR check and it was added ~5 years ago when I first started pushing on determinism in the compiler. This flag just writes out the deterministic key file next to the binary. When the deterministic check fails it allows the developer to diff the key files and determine if the compilation was given different inputs. If there are different inputs then determinism failed due to a build system bug, if it's the same inputs then it's a compiler bug. 

This has saved me and others massive amounts of time over the years.

Reply:
I'm not questioning this feature flag :-) My observation is just about lack of test coverage.

Reply:
Gotcha. Yes this code was added ~5 years ago and had no test coverage. That is why no tests failed as a result of this. Can try and add some test coverage here (generally this path is awkward to test which is likely why it wasn't tested in the first place).

Reply:
Spent some time this morning adding a test case here. I focused on the fact that it is hard to write tests that cover this code path. Hence spent some time factoring out a bit of code to make it easier. It looks promising and should be as easy as the following when I am done.

```c#
var (compiler, fileSystem) = CSharpCompilerUtil.CreateForNetCoreApp("test.cs");
compiler.Run();
````

The change is a bit invasive though. Do you want me to keep building on this PR or finish of this PR and submit the refactor + test of this code path as a sep change? Fine with either.

Reply:
Thanks for looking into it. Since this is not central to current PR, let's punt.

Reply:
I went ahead and put it in the current PR. When working on the change I found a bug in this path that I wanted to fix. I put it as a separate commit to make it easier to review.

Reply:
Just curious, where was the bug?

Reply:
There were actually two bugs: 

1. The placement of the feature flag check meant that it only ran if the compilation had analyzers or generators. 
2. The original `Arguments.EmitOptions` was being used instead of the local `emitOptions` which meant it didn't have items like the `pdbPath` set.